ui/src/app/components/workflows/workflow/dynamic-parts/cron-quartz-part/cron-quartz-part.component.ts

```diff
@@ -16,198 +32,73 @@ export class CronQuartzPartComponent implements OnInit, AfterViewInit {
-      label: 'Tuesday',
-    },
-    {
-      value: 3,
-      label: 'Wednesday',
-    },
-    {
-      value: 4,
-      label: 'Thursday',
-    },
-    {
-      value: 5,
-      label: 'Friday',
-    },
-    {
-      value: 6,
-      label: 'Saturday',
-    },
-    {
-      value: 7,
-      label: 'Sunday',
-    },
-  ];
-  dayOfMonthValues = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31];
-  monthValues = [
-    {
-      value: 1,
-      label: 'January',
-    },
-    {
-      value: 2,
-      label: 'February',
-    },
-    {
-      value: 3,
-      label: 'March',
-    },
-    {
-      value: 4,
-      label: 'April',
-    },
-    {
-      value: 5,
-      label: 'May',
-    },
-    {
-      value: 6,
-      label: 'June',
-    },
-    {
-      value: 7,
-      label: 'July',
-    },
-    {
-      value: 8,
-      label: 'August',
-    },
-    {
-      value: 9,
-      label: 'September',
-    },
-    {
-      value: 10,
-      label: 'October',
-    },
-    {
-      value: 11,
-      label: 'November',
-    },
-    {
-      value: 12,
-      label: 'December',
-    },
-  ];
-  hourValues = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23];
-  minuteValues = [0, 5, 10, 15, 20, 25, 30, 35, 40, 45, 50, 55];
-  minutesValues = [5, 10, 15, 20, 25, 30, 35, 40, 45, 50, 55];
+  hourValues = cronExpressionOptions.HOUR_VALUES;
+  frequencies = cronExpressionOptions.FREQUENCIES;
+  minuteValues = cronExpressionOptions.MINUTE_VALUES;
+  minutesValues = cronExpressionOptions.MINUTES_VALUES;
+  cron: string[] = [];
 
   constructor() {
-    // do nothing
   }
 
   ngOnInit(): void {
-    this.fromCron(this.value);
   }
 
   ngAfterViewInit(): void {
-    // this.fromCron(this.value)
   }
 
-  setCron(n): string {
-    const cron = ['0', '*', '*', '*', '*', '?'];
-
-    if (n && n.base && n.base >= 2) {
-      cron[1] = typeof n.minuteValue !== undefined ? n.minuteValue : '0';
-    }
-
-    if (n && n.base && n.base >= 3) {
-      cron[2] = typeof n.hourValue !== undefined ? n.hourValue : '*';
-    }
-
-    if (n && n.base && n.base === 4) {
-      cron[3] = '?';
-      cron[5] = n.dayValue;
-    }
-
-    if (n && n.base && n.base >= 5) {
-      cron[3] = typeof n.dayOfMonthValue !== undefined ? n.dayOfMonthValue : '?';
+  setCron(value, label) {
+    this.cron = ['0', '0', '0', '?', '*', '*', '*'];
+    switch (label) {
+      case 'Hour every':
+        this.cron[1] = `0/${value}`;
+        break;
+      case 'Hour at':
+        this.cron[1] = value;
+        break;
+      case 'Day':
+        this.cron[2] = value;
+        break;
+      default:
+        this.cron = ['0', '0', '0', '?', '*', '*', '*'];
+        break;
     }
-
-    if (n && n.base && n.base === 6) {
-      cron[4] = typeof n.monthValue !== undefined ? n.monthValue : '*';
-    }
-    return cron.join(' ');
+    this.modelChanged();
   }
 
   fromCron(value: string) {
```

Comment:
This code is never called. I think you should use something like this to fill the values for `minuteValue`, `hourValue`, `dayValue` when the input value changes. Btw how to intercept input property: https://angular.io/guide/component-interaction#intercept-input-property-changes-with-a-setter

Reply:
the concept around this method is getting a Cron expression from the back-end, do some manipulations and then show it to the UI ( the way I understood it ). Leaving it as-is for now.